cluster-autoscaler/cloudprovider/digitalocean/digitalocean_node_group.go

```diff
@@ -233,9 +234,12 @@ func (n *NodeGroup) Autoprovisioned() bool {
 // toInstances converts a slice of *godo.KubernetesNode to
 // cloudprovider.Instance
 func toInstances(nodes []*godo.KubernetesNode) []cloudprovider.Instance {
-	instances := make([]cloudprovider.Instance, len(nodes))
-	for i, nd := range nodes {
-		instances[i] = toInstance(nd)
+	instances := []cloudprovider.Instance{}
+	for _, nd := range nodes {
+		if nd.Status == nil {
```

Comment:
When do we expect a `godo.KubernetesNode` status to be nil and why are we ignoring nodes with that status in this PR?

Reply:
Looking at the API code it does seem like it should never be `nil`, but it is defined as a pointer to the struct so it's possible for it to be in the future if that code changes. There was also an [existing test case](https://github.com/digitalocean/autoscaler/blob/master/cluster-autoscaler/cloudprovider/digitalocean/digitalocean_node_group_test.go#L273-L274) for `no status` that I didn't want to break/remove. I think it could be argued that since the code today seems to always populate `NodeStatus` then we don't need this, but since this is a decentralized component I'd probably ere on keeping it.

Reply:
I guess part of my confusion about handling it here is that it appears to [already be handled in toInstanceStatus](https://github.com/digitalocean/autoscaler/pull/2/files#diff-e0b7c524d95e8a20f927ee36bc4b0ce6R264)

Reply:
Good point, will fix this :+1:

Reply:
https://github.com/digitalocean/autoscaler/pull/2/commits/f0a2cecc6d9e968542931fe6f1e1b4fae8a9649a